snapcraft/internal/pluginhandler.py

```diff
@@ -322,6 +322,13 @@ def clean_build(self, hint=''):
     def migratable_fileset_for(self, step):
         plugin_fileset = self.code.snap_fileset()
         fileset = (getattr(self.code.options, step, ['*']) or ['*']).copy()
+        # If we're priming and we don't have an explicit set of files to prime
+        # include the files from the stage step
+        if (step == 'prime' or step == 'snap') and fileset == ['*']:
+            stage_fileset = (
+                getattr(self.code.options, 'stage', ['*']) or ['*']).copy()
```

Comment:
We use this in a few different places, and it's a little tough to read. Can we extract this logic into a less compact helper function?